### ISSUE TITLE
Extension of internal protocol cannot be declared public

### ISSUE THREAD

**Issue by MaxDesiatov** [nicklockwood/SwiftFormat#794]
Changes applied by the latest SwiftFormat break the build with public extensions of internal protocols. See https://github.com/TokamakUI/Tokamak/commit/33adba20ab145bbe3277085324f7343d11c2763a that had to be applied to fix the build. 

The gist is that some functions on extensions of private protocols need to be public to satisfy requirements of other public protocols. Consider this:

```swift
internal protocol P1 {
}

extension P1 {
  public func f() {}
}

public protocol P2 {
  public func f()
}

public struct S: P1, P2 {}
```

SwiftFormat will break this code as it will make the extension on `P1` public, which will cause the `Extension of internal protocol cannot be declared public` compilation error.

**Comment by nicklockwood** [comment]
@MaxDesiatov thanks for reporting. Looks like this is already fixed on the develop branch.

**Comment by nicklockwood** [comment]
@MaxDesiatov fixed in 0.47.3

**Comment by MaxDesiatov** [comment]
This is great, thank you!
